packages/jellyfish-api-core/src/category/rawtx.ts

```diff
@@ -0,0 +1,202 @@
+import { BigNumber, ApiClient } from '../.'
+
+export enum SigHashType {
+  ALL = 'ALL',
+  NONE = 'NONE',
+  SINGLE = 'SINGLE',
+  ALL_ANYONECANPAY = 'ALL|ANYONECANPAY',
+  NONE_ANYONECANPAY = 'NONE|ANYONECANPAY',
+  SINGLE_ANYONECANPAY = 'SINGLE|ANYONECANPAY',
+}
+
+/**
+ * RawTransaction RPCs for DeFi Blockchain
+ */
+export class RawTx {
+  private readonly client: ApiClient
+
+  constructor (client: ApiClient) {
+    this.client = client
+  }
+
+  /**
+   * Create a transaction spending the given inputs and creating new outputs that returns a hex-encoded raw transaction.
+   * Note that the transaction's inputs are not signed, and it is not stored in the wallet or transmitted to the network.
+   *
+   * @param {CreateRawTxIn[]} inputs array of inputs
+   * @param {CreateRawTxOut[]} outputs array with outputs
+   * @param {CreateRawTxOptions} options
+   * @param {number} options.locktime Non-0 value also locktime-activates inputs
+   * @param {boolean} options.replaceable Marks this transaction as BIP125-replaceable
+   * @return {Promise<string>} hex string of the transaction
+   */
+  async createRawTransaction (
+    inputs: CreateRawTxIn[],
+    outputs: CreateRawTxOut,
+    options: CreateRawTxOptions = {}
+  ): Promise<string> {
+    const { locktime = 0, replaceable = false } = options
+
+    return await this.client.call('createrawtransaction', [
+      inputs, outputs, locktime, replaceable
+    ], 'number')
+  }
+
+  /**
+   * Sign inputs for raw transaction (serialized, hex-encoded), Providing an array of base58-encoded private keys that
+   * will be the keys used to sign the transaction. An optional array of previous transaction outputs that this
+   * transaction depends on but may not yet be in the blockchain.
+   *
+   * @param {string} rawTx unsigned raw transaction
+   * @param {string[]} privKeys array of base58-encoded private keys for signing (WIF)
+   * @param {SignRawTxWithKeyPrevTx[]} prevTxs array of previous dependent transaction outputs
+   * @param {SignRawTxWithKeyOptions} options
+   * @param {SigHashType} options.sigHashType The signature hash type to use
+   * @return {Promise<SignRawTxWithKeyResult>}
+   */
+  async signRawTransactionWithKey (
+    rawTx: string,
+    privKeys: string[],
+    prevTxs?: SignRawTxWithKeyPrevTx[],
+    options: SignRawTxWithKeyOptions = {}
```

Comment:
is there any reason put the `prevTxs` out of options??
else we follow back the deno style
`Exported functions: max 2 args, put the rest into an options object.`
https://github.com/denoland/deno/blob/main/docs/contributing/style_guide.md